cinder/tests/unit/volume/drivers/nexenta/test_nexenta5_jsonrpc.py

```diff
@@ -0,0 +1,245 @@
+# Copyright 2018 Nexenta Systems, Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+Unit tests for NexentaStor 5 REST API helper
+"""
+
+import mock
+import requests
+import uuid
+
+from cinder import exception
+from cinder import test
+from cinder.volume.drivers.nexenta.ns5 import jsonrpc
+from mock import patch
+from oslo_serialization import jsonutils
+from requests import adapters
+
+HOST = '1.1.1.1'
+USERNAME = 'user'
+PASSWORD = 'pass'
+
+
+def gen_response(code=200, json=None):
+    r = requests.Response()
+    r.headers['Content-Type'] = 'application/json'
+    r.encoding = 'utf8'
+    r.status_code = code
+    r.reason = 'FAKE REASON'
+    r.raw = mock.Mock()
+    r._content = ''
+    if json:
+        r._content = jsonutils.dumps(json)
+    return r
+
+
+class TestNexentaJSONProxyAuth(test.TestCase):
+
+    @patch('cinder.volume.drivers.nexenta.ns5.jsonrpc.requests.post')
+    def test_https_auth(self, post):
+        use_https = True
+        port = 8443
+        auth_uri = 'auth/login'
+        rnd_url = 'some/random/url'
+        ssl = False
+
+        class PostSideEffect(object):
+            def __call__(self, *args, **kwargs):
+                r = gen_response()
+                if args[0] == '{}://{}:{}/{}'.format(
+                        'https', HOST, port, auth_uri):
+                    token = uuid.uuid4().hex
+                    content = {'token': token}
+                    r._content = jsonutils.dumps(content)
+                return r
+        post_side_effect = PostSideEffect()
+        post.side_effect = post_side_effect
```

Comment:
What is the purpose of the previous assignment that is overwritten with this line?
What is class PostSideEffect for? I do not see it is used anywhere except one usage above.